src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -333,7 +333,7 @@ private string GenerateForTypeWithUnknownConverter(TypeGenerationSpec typeMetada
                     metadataInitSource.Append($@"
                         if (!converter.CanConvert(typeToConvert))
                         {{
-                            throw new {InvalidOperationExceptionTypeRef}($""The converter '{{converter.GetType()}}' is not compatible with the type '{{typeToConvert}}'."");
+                            throw new {InvalidOperationExceptionTypeRef}(string.Format(""{SR.Exception_IncompatibleConverterType}"", converter.GetType(), typeToConvert));
```

Comment:
With this, we are localizing run-time messages in the generated code, even though we don't localize other run-time exceptions thrown from the runtime AFAIK. Do other generators do this?

My previous thoughts on this we that we shouldn't localize run-time exceptions since we don't do this elsewhere; we should only localize messages shown within VS as errors\warnings. If \ when we decide to localize messages in the runtime, then I think that would be the best time to do this for generated code. Localizing runtime messages may not always be desired -- it does make it harder to search for the message, and increases download size.

Also we generate the message with the current VS locale. Is that what developers want? They may want the message of the current locale that is running and not the locale that it was developed with.

Reply:
How can we keep the exception messages in resource strings without getting them localized?

Reply:
Since the strings shouldn't be localized, keeping them in code is fine IMHO. Perhaps move all such strings to a common location as `const string` and document this there, which may also help moving those to resources in the future.

However, using 2 resx files with 1 as en-US.resx along with a different resource manager instance that only uses en-US should work too.

Reply:
In the context of this backporting PR what is the preferred course of action?

Reply:
FWIW in the original PR I neglected that a reference to `SR.SomeString` would yield a localized string since it is evaluated within the generator's runtime execution.

Reply:
I've pushed changes moving the messages into const string in source. PTAL.